src/pages/Background/index.ts

```diff
@@ -270,3 +270,26 @@ browser.omnibox.onInputEntered.addListener(async (content: string, disposition:
 });
 
 
+// refresh for sidepanel on tab change
+chrome.tabs.onUpdated.addListener((tabId, changeInfo) => {
+  // if (changeInfo.url) {
+    console.log("Tab URL changed:", changeInfo.url, " Tab ID:", tabId);
+
+    // Reload the side panel if needed
+    chrome.sidePanel.setOptions({ path: "./index.html" });
+  // }
+});
+
+// Listen for tab changes 
+chrome.tabs.onActivated.addListener(async (activeInfo) => {
+  console.log("Tab changed:", activeInfo);
+
+  try {
+    await chrome.sidePanel.setOptions({
+      path: "./index.html"
+    });
+    console.log("Side panel reloaded.");
+  } catch (error) {
+    console.error("Failed to reload the side panel:", error);
+  }
+});
```

Comment:
ummmm.... i inspected through the background.js console it does execute (see attached image) but `chrome.sidePanel.setOptions()` function doesn't reload the tab if it's already set to that for that we need to separate the sidepanel page and still i don't know vite so but i'll try to learn vite and implement it via `chrome.runtime.onMessage.addListener()` which should be able to execute the reload code `window.location.reload();` 
![image](https://github.com/user-attachments/assets/bbed31e9-bcb6-4c55-ac4e-2b8b2556d2ca)